libtiledbsoma/src/soma/soma_group.cc

```diff
@@ -67,6 +67,14 @@ std::unique_ptr<SOMAGroup> SOMAGroup::create(
             static_cast<uint32_t>(ENCODING_VERSION_VAL.length()),
             ENCODING_VERSION_VAL.c_str());
 
+        // Root SOMA objects include a `dataset_type` entry to allow the
+        // TileDB Cloud UI to detect that they are SOMA datasets.
+        std::string cloud_scheme = "tiledb://";
```

Comment:
Let's go ahead & put this regardless of URI -- as it was before. People can ingest directly to TileDB Cloud, or, ingest to S3 and register it.

There's no harm in writing it unconditionally, and value to be gained.